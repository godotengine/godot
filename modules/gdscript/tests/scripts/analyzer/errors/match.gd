func test():
	var dict = { a = 1 }
	var a = 1
	match 2:
		dict["a"]: # TODO: Fix positional information (parser bug).
			print("not allowed")
		a + 2:
			print("not allowed")
		_ when b == 0:
			print("b does not exist")
