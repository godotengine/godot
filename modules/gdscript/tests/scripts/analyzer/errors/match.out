GDTEST_ANALYZER_ERROR
>> ERROR at line 7: Expression in match pattern must be a constant expression, an identifier, or an attribute access ("A.B").
>> ERROR at line 9: Identifier "b" not declared in the current scope.
>> ERROR at line 12: Expression in match pattern must be a constant expression, an identifier, or an attribute access ("A.B").
